aten/src/ATen/native/cuda/Activation.cu

```diff
@@ -297,9 +297,9 @@ void softplus_backward_kernel(TensorIterator& iter, const Scalar& beta_, const S
   AT_DISPATCH_FLOATING_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, iter.dtype(), "softplus_backward_cuda", [&]() {
     auto beta = beta_.to<scalar_t>();
     auto threshold = threshold_.to<scalar_t>();
-    gpu_kernel(iter, [beta, threshold]GPU_LAMBDA(scalar_t a, scalar_t b) -> scalar_t {
+    gpu_kernel(iter, [beta, threshold]GPU_LAMBDA(scalar_t a, scalar_t b, scalar_t c) -> scalar_t {
```

Comment:
Just a quick thought, a possible way around having to introduce `ternary_op` is to pass `in` instead of `out` as b, and then the below line becomes:
```cpp
 return (b * beta) > threshold ? a : a * z / (z + scalar_t(1.)) ;
```

Reply:
@shmsong I thought about doing this originally, but it gives incorrect results (note that `z` is computed from `b`, so its value changes if `b` changes).

Both input and output need to be available to the backward pass computation- input is used for the threshold and output is used for the value itself.

Reply:
Yes `z` will be different and the part using `z` also needs to be updated: (I should have highlighted this part in the original comment)
```
a * (z - scalar_t(1.)) / z
``` 
-> 
```
a * z / (z + scalar_t(1.))
```
There is a useful identity on softplus:
```
 exp(beta*out) == exp(beta*in) + 1
```
[https://pytorch.org/docs/stable/generated/torch.nn.Softplus.html](https://pytorch.org/docs/stable/generated/torch.nn.Softplus.html)

This should give correct result if I understand it correctly.

Reply:
Ah yes, thanks for clarifying. Looks right to me and avoids the addition of `ternary_op`.